phidgets/lcd.go

```diff
@@ -10,148 +10,41 @@ void ccallback(void* handle, void* ctx, double b);  // Forward declaration.
 */
 import "C"
 import (
-	"errors"
-	"reflect"
 	"unsafe"
 )
 
 //PhidgetLCD is the struct that is a phidget lcd sensor
 type PhidgetLCD struct {
+	phidget
 	handle C.PhidgetLCDHandle
 }
 
 //Create creates a phidget lcd sensor
 func (p *PhidgetLCD) Create() {
 	C.PhidgetLCD_create(&p.handle)
+	p.rawHandle(unsafe.Pointer(p.handle))
 }
 
 //SetText sets the lcd text
-func (p *PhidgetLCD) SetText(text string) {
-	C.PhidgetLCD_writeText(p.handle, C.FONT_6x12, 40, 25, stringToCCharArray(text))
-	C.PhidgetLCD_flush(p.handle)
-}
-
-//SetBacklight - sets the backlight value
-func (p *PhidgetLCD) SetBacklight(brightness float32) {
-	C.PhidgetLCD_setBacklight(p.handle, float32ToCdouble(brightness))
-}
-
-//Common to all derived phidgets
-
-func (p *PhidgetLCD) getErrorDescription(cerr C.PhidgetReturnCode) string {
-	var errorString *C.char
-	C.Phidget_getErrorDescription(cerr, &errorString)
-	//Get the name of our class
-	t := reflect.TypeOf(p)
-	return t.Elem().Name() + ": " + C.GoString(errorString)
-}
-
-//SetIsRemote sets a phidget sensor as a remote device
-func (p *PhidgetLCD) SetIsRemote(b bool) error {
-	h := (*C.struct__Phidget)(unsafe.Pointer(p.handle))
-	cerr := C.Phidget_setIsRemote(h, boolToCInt(b))
-	if cerr != C.EPHIDGET_OK {
-		return errors.New(p.getErrorDescription(cerr))
-	}
-	return nil
-
-}
-
-//SetDeviceSerialNumber sets a phidget lcd sensor's serial number
-func (p *PhidgetLCD) SetDeviceSerialNumber(serial int) error {
-	h := (*C.struct__Phidget)(unsafe.Pointer(p.handle))
-	cerr := C.Phidget_setDeviceSerialNumber(h, intToCInt(serial))
-	if cerr != C.EPHIDGET_OK {
-		return errors.New(p.getErrorDescription(cerr))
-	}
-	return nil
-}
-
-//SetHubPort sets a phidget lcd sensor's hub port
-func (p *PhidgetLCD) SetHubPort(port int) error {
-	h := (*C.struct__Phidget)(unsafe.Pointer(p.handle))
-	cerr := C.Phidget_setHubPort(h, intToCInt(port))
+func (p *PhidgetLCD) SetText(text string) error {
+	str := C.CString(text)
+	cerr := C.PhidgetLCD_writeText(p.handle, C.FONT_6x12, 40, 25, str)
```

Comment:
Note to self: I'd like to break out the `writeText` options in addition to this basic `SetText`. Better suited for a follow-up PR.